src/useGraph.ts

```diff
@@ -106,15 +145,31 @@ export const useGraph = ({
       });
 
       // Calculate clusters
-      const clusters = calculateClusters({
+      const newClusters = calculateClusters({
         nodes: result.nodes,
         clusterAttribute
       });
 
+      if (constrainDragging) {
```

Comment:
I'm not sure I follow what this is for when using constrained dragging - need to add comments.

Reply:
comment was present few lines below, but I moved it to above for full code block